src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -680,6 +684,8 @@ discover_dfuse_mounts(void)
 			atomic_init(&pt_dfs_mt->inited, 0);
 			pt_dfs_mt->pool         = NULL;
 			D_STRNDUP(pt_dfs_mt->fs_root, fs_entry->mnt_dir, pt_dfs_mt->len_fs_root);
+			if (pt_dfs_mt->fs_root == NULL)
+				D_STRNDUP(pt_dfs_mt->fs_root, fs_entry->mnt_dir, pt_dfs_mt->len_fs_root);
```

Comment:
Same comment as above, although in this case the error handling in the function will also be called.

Reply:
I removed the retry of D_STRNDUP() here.